optuna/samplers/_tpe/parzen_estimator.py

```diff
@@ -216,6 +214,15 @@ def _calculate_weights(self, predetermined_weights: Optional[np.ndarray]) -> np.
 
         if predetermined_weights is None:
             w = weights_func(n_observations)[:n_observations]
+            if w is not None:
+                if np.any(w < 0):
+                    raise ValueError("Negative parameters.weights are not allowed.")
+                if len(w) > 0 and np.sum(w) <= 0:
+                    raise ValueError("parameters.weights of all zeros are not allowed")
+                if not np.all(np.isfinite(w)):
+                    raise ValueError(
+                        "parameters.weights of infinite or NaN values are not allowed."
+                    )
```

Comment:
How about returning more descriptive error message as follows?
```suggestion
                if np.any(w < 0):
                    raise ValueError(f"The `weights` function is not allowed to return negative values {w}. The argument of the `weights` function is {n_observations}.")
                if len(w) > 0 and np.sum(w) <= 0:
                    raise ValueError(f"The `weight` function is not allowed to return all-zero values {w}. The argument of the `weights` function is {n_observations}.")
                if not np.all(np.isfinite(w)):
                    raise ValueError(
                        f"The `weights`function is not allowed to return infinite or NaN values {w}. The argument of the `weights` function is {n_observations}."
                    )
```